source/gui/addonGui.py

```diff
@@ -5,128 +5,344 @@
-		entryButtonsSizer.Add(self.removeButton)
+		self.removeButton.Bind(wx.EVT_BUTTON, self.onRemoveClick)
+		listAndButtonsSizerHelper.addItem(entryButtonsHelper.sizer)
+
+		mainSizer.Add(
+			listAndButtonsSizerHelper.sizer,
+			border=guiHelper.BORDER_FOR_DIALOGS,
+			flag=wx.ALL
+		)
+
+		# the following buttons are more general and apply regardless of the current selection.
+		generalActions=guiHelper.ButtonHelper(wx.HORIZONTAL)
 		# Translators: The label of a button in Add-ons Manager to open the Add-ons website and get more add-ons.
-		self.getAddonsButton=wx.Button(self,label=_("&Get add-ons..."))
-		self.getAddonsButton.Bind(wx.EVT_BUTTON,self.onGetAddonsClick)
-		entryButtonsSizer.Add(self.getAddonsButton)
-		settingsSizer.Add(entryButtonsSizer)
-		mainSizer.Add(settingsSizer,border=20,flag=wx.LEFT|wx.RIGHT|wx.TOP)
+		self.getAddonsButton = generalActions.addButton(self, label=_("&Get add-ons..."))
+		self.getAddonsButton.Bind(wx.EVT_BUTTON, self.onGetAddonsClick)
+		# Translators: The label for a button in Add-ons Manager dialog to install an add-on.
+		self.addButton = generalActions.addButton(self, label=_("&Install..."))
+		self.addButton.Bind(wx.EVT_BUTTON, self.onAddClick)
+		# Translators: The label of a button in the Add-ons Manager to open the list of incompatible add-ons.
+		self.incompatAddonsButton = generalActions.addButton(self, label=_("&Manage incompatible add-ons..."))
+		self.incompatAddonsButton.Bind(wx.EVT_BUTTON, self.onIncompatAddonsShowClick)
+
+		mainSizer.Add(
+			generalActions.sizer,
+			border=guiHelper.BORDER_FOR_DIALOGS,
+			flag=wx.LEFT | wx.RIGHT
+		)
+
+		mainSizer.Add(
+			wx.StaticLine(self),
+			border=guiHelper.BORDER_FOR_DIALOGS,
+			flag=wx.TOP | wx.BOTTOM | wx.EXPAND
+		)
+
 		# Translators: The label of a button to close the Addons dialog.
 		closeButton = wx.Button(self, label=_("&Close"), id=wx.ID_CLOSE)
 		closeButton.Bind(wx.EVT_BUTTON, lambda evt: self.Close())
-		mainSizer.Add(closeButton,border=20,flag=wx.LEFT|wx.RIGHT|wx.BOTTOM|wx.CENTER|wx.ALIGN_RIGHT)
+		mainSizer.Add(
+			closeButton,
+			border=guiHelper.BORDER_FOR_DIALOGS,
+			flag=wx.LEFT | wx.RIGHT | wx.BOTTOM | wx.CENTER | wx.ALIGN_RIGHT
+		)
 		self.Bind(wx.EVT_CLOSE, self.onClose)
 		self.EscapeId = wx.ID_CLOSE
+
 		mainSizer.Fit(self)
 		self.SetSizer(mainSizer)
 		self.refreshAddonsList()
 		self.addonsList.SetFocus()
 		self.CentreOnScreen()
 
-	def onAddClick(self,evt):
+	def onAddClick(self, evt):
 		# Translators: The message displayed in the dialog that allows you to choose an add-on package for installation.
-		fd=wx.FileDialog(self,message=_("Choose Add-on Package File"),
+		fd = wx.FileDialog(self, message=_("Choose Add-on Package File"),
 		# Translators: the label for the NVDA add-on package file type in the Choose add-on dialog.
 		wildcard=(_("NVDA Add-on Package (*.{ext})")+"|*.{ext}").format(ext=addonHandler.BUNDLE_EXTENSION),
-		defaultDir="c:",style=wx.FD_OPEN)

```

Comment:
I wonder whether %userprofile% makes more sense as default dir

Reply:
Maybe, but I don't think I'll address that in this PR